Tree/patch 1

TobiasZawada opened the pull request:
### Brief summary of what the package does

Add AUCTeX preview to non-LaTeX buffers.
Examples:
 - html pages with MathJax formulas
 - source code with equations in doxygen comments
 - stackexchange pages with equations viewed with package sx.el

### Direct link to the package repository

https://github.com/TobiasZawada/texfrag

### Your association with the package

maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

purcell commented:
Looks good, thanks. You should probably add some `;;;###autoload` cookie comments for the mode definition forms and/or `texfrag-org-mode-hook-function`.

TobiasZawada commented:
@purcell Thank you very much for your time. Your advices are very valuable. You are clearly right about the autoload cookie comments and I will review the `texfrag-org-mode-hook-function` thing.

Sorry for the first failing commit. I had to do everything in the time that was left after I had to care for my 19 month year old son and my wife both currently fighting with a cold.

purcell commented:
> Sorry for the first failing commit. I had to do everything in the time that was left after I had to care for my 19 month year old son and my wife both currently fighting with a cold.

No worries at all - I know how that goes. Thanks for sharing your work.

TobiasZawada commented:
I've added the autoloads and replaced the wrong `texfrag-org-mode-hook-function` with the right `texfrag-org` in the comment.
I've also extended the comments, clearly separated the sections for the supported major modes and added an option `texfrag-preview-buffer-at-start` that does exactly what it says. Enabling this option turns out to be really nice in `sx-question-mode`. One just has to open some question from `math.stackexchange.com` with `sx.el` and the equations are already rendered by LaTeX. Right so as one would open the page with firefox and MathJax support.
Best regards and Happy New Year!
